lib/xdrgen/generators/rust/src/types.rs

```diff
@@ -1219,8 +1222,15 @@ impl<const MAX: u32> ReadXdr for VecM<u8, MAX> {
                 return Err(Error::LengthExceedsMax);
             }
 
-            let mut vec = vec![0u8; len as usize];
-            r.read_exact(&mut vec)?;
+            let mut vec = vec![0u8; 0];
+            let mut len_remaining = len as usize;
+            while len_remaining > 0 {
+                let len_read = core::cmp::min(len_remaining, MAX_PREALLOCATED_BYTES_READ);
+                let offset = vec.len();
+                vec.resize(vec.len() + len_read, 0);
```

Comment:
maybe I'm not following the goal here, but wouldn't these resize calls defeat the purpose of this PR ?

Reply:
The resize calls rather than preallocating the entire buffer up front defer the allocation until we've been able to parse some of the input.

For example, for a 2048 byte input, the logic will allocate 1024 memory, and then parse the first 1024 bytes before allocating more memory.